### PR TITLE
transition part of http_parser to use span

### PR DESCRIPTION
**Pull request by arvidn**

### CODE REVIEW

File: src/http_connection.cpp [hunk 1]
```diff
@@ -743,10 +743,9 @@ void http_connection::on_read(error_code const& e
 
 	if (m_bottled || !m_parser.header_finished())
 	{
-		libtorrent::buffer::const_interval rcv_buf(&m_recvbuffer[0]
-			, &m_recvbuffer[0] + m_read_pos);
+		span<char const> rcv_buf(m_recvbuffer);
 		bool error = false;
-		m_parser.incoming(rcv_buf, error);
+		m_parser.incoming(rcv_buf.subspan(0, m_read_pos), error);
```
Comment (aldenml) [inline src/http_connection.cpp]: I wonder if `rcv_buf` could be eliminated

File: src/http_parser.cpp [hunk 2]
```diff
@@ -364,7 +348,8 @@ namespace libtorrent
 						std::get<0>(ret) += int(payload);
 						incoming -= int(payload);
 					}
-					buffer::const_interval buf(recv_buffer.begin + m_cur_chunk_end, recv_buffer.end);
+					span<char const> buf(recv_buffer.data() + m_cur_chunk_end
```
Comment (aldenml) [inline src/http_parser.cpp]: can `span:: last` be used here?

File: src/http_connection.cpp (hunk 1 above)
Comment (arvidn) [inline src/http_connection.cpp]: if there was a free function version of subspan() m_recvbuf could be passed in directly, but there isn't one right now

File: src/http_parser.cpp (hunk 2 above)
Comment (arvidn) [inline src/http_parser.cpp]: subspan() can be used

**Pull request merged**